arm-customer-insights/2017-04-26/swagger/customer-insights.json

```diff
@@ -0,0 +1,5664 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/kpi/{kpiName}": {
+      "put": {
+        "tags": [
+          "Kpi"
+        ],
+        "operationId": "Kpi_CreateOrUpdate",
+        "x-ms-examples": {
+          "Kpi_CreateOrUpdate": {
+            "$ref": "../examples/KpiCreateOrUpdate.json"
+          }
+        },
+        "description": "Creates a KPI or updates an existing KPI in the hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "kpiName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the KPI.",
+            "pattern": "^[a-zA-Z][a-zA-Z0-9_]+$",
+            "minLength": 1,
+            "maxLength": 512
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KpiResourceFormat"
+            },
+            "description": "Parameters supplied to the create/update KPI operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully created the KPI.",
+            "schema": {
+              "$ref": "#/definitions/KpiResourceFormat"
+            }
+          },
+          "202": {
+            "description": "Accepted. The create KPI operation is accepted."
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "get": {
+        "tags": [
+          "Kpi"
+        ],
+        "operationId": "Kpi_Get",
+        "x-ms-examples": {
+          "Kpi_Get": {
+            "$ref": "../examples/KpiGet.json"
+          }
+        },
+        "description": "Gets a KPI in the hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "kpiName",
+            "in": "path",
+            "requir
```

Comment:
This resource type name conflicts with the Microsoft.Authorization/roleAssignments which is a commonly known RBAC model. This should not have been named this since we dont want to confuse users with the Azure wide concepts like RBAC role assignments.